pybop/costs/error_costs.py

```diff
@@ -38,7 +38,14 @@ def __call__(self, x, grad=None):
         Computes the cost.
         """
         try:
-            return np.sqrt(np.mean((self.problem.evaluate(x) - self._target) ** 2))
+            prediction = self.problem.evaluate(x)
+
+            if len(prediction) < len(self._target):
+                cost = np.Infinity  # simulation stopped early
```

Comment:
np.inf is probably a better constant to use: https://numpy.org/doc/stable/reference/constants.html

Reply:
Thanks for the link!